service/src/parentchain_config.rs

```diff
@@ -0,0 +1,140 @@
+/*
+	Copyright 2021 Integritee AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+use ita_parentchain_interface::{integritee, target_a, target_b};
+use itc_parentchain::primitives::ParentchainId;
+use itp_node_api::{
+	api_client::{Api, ApiClientError, ApiResult, AssetRuntimeConfig, StorageKey},
+	node_api_factory::{NodeApiFactory, NodeApiFactoryError, ParentchainApiWrapper},
+};
+use itp_types::parentchain::Hash;
+use sp_core::sr25519;
+use substrate_api_client::{
+	ac_primitives::{Bytes, WithExtrinsicParams},
+	rpc::{Request, TungsteniteRpcClient},
+	ExtrinsicReport, GetStorage, ReadProof, SubmitAndWatch, SubmitExtrinsic, XtStatus,
+};
+
+/// Standard runtime config for Substrate and Polkadot nodes.
+pub type IntegriteeRuntimeConfig =
+	WithExtrinsicParams<AssetRuntimeConfig, integritee::ParentchainExtrinsicParams>;
+
+/// runtime config for chains like Asset Hub or Encointer
+pub type TargetARuntimeConfig =
+	WithExtrinsicParams<AssetRuntimeConfig, target_a::ParentchainExtrinsicParams>;
+
+/// Standard runtime config for Substrate and Polkadot nodes.
+pub type TargetBRuntimeConfig =
+	WithExtrinsicParams<AssetRuntimeConfig, target_b::ParentchainExtrinsicParams>;
+
+pub enum ParentchainApiLocal {
```

Comment:
this I somehow like and it seems to work for OCall. Maybe we could make it work for service-main instead of the `ParentchainApiWrapper` shizzle?

Reply:
Yes, I think this is the way to go actually for many things, also when we distinguish build flavors.